webpay/pin/api.py

```diff
@@ -0,0 +1,78 @@
+from django.http import Http404
+from django.core.exceptions import ObjectDoesNotExist
+
+from rest_framework import response, serializers, viewsets
+
+from lib.solitude.api import client
+from webpay.api.base import BuyerIsLoggedIn
+from webpay.api.utils import api_error
+from webpay.auth.utils import set_user_has_pin
+from webpay.pin.forms import CreatePinForm, ResetPinForm, VerifyPinForm
+
+
+class PinSerializer(serializers.Serializer):
+    pin = serializers.BooleanField(default=False)
+    pin_locked_out = serializers.DateTimeField()
+    pin_is_locked_out = serializers.BooleanField(default=False)
+    pin_was_locked_out = serializers.BooleanField(default=False)
+    pin_reset_started = serializers.BooleanField(default=False)
+
+
+class PinViewSet(viewsets.ViewSet):
+    permission_classes = (BuyerIsLoggedIn,)
+    serializer_class = PinSerializer
+
+    def retrieve(self, request):
+        res = client.get_buyer(request.session['uuid'])
+        if res:
+            res['pin_reset_started'] = request.session.get(
+                'was_reverified', False)
```

Comment:
This isn't being set anymore as far as I can tell. I think you'll want a helper with the logic from the `ResetPinForm` to check the start and auth timestamps and set this to `True`/`False` accordingly.